skrifa/src/outline/mod.rs

```diff
@@ -0,0 +1,448 @@
+    ///
+    /// The font size, location in variation space and hinting mode are
+    /// defined by the current configuration of the given hinting instance.
+    pub fn native_hinted(instance: &'a NativeHintingInstance) -> Self {
+        Self {
+            instance: DrawInstance::NativeHinted(instance),
+            memory: None,
+        }
+    }
+
+    /// Builder method to associate a user memory buffer to be used for
+    /// temporary allocations during drawing.
+    ///
+    /// The required size of this buffer can be computed using the
+    /// [`OutlineGlyph::draw_memory_size`] method.
+    ///
+    /// If not provided, any necessary memory will be allocated internally.
+    pub fn with_memory(mut self, memory: Option<&'a mut [u8]>) -> Self {
+        self.memory = memory;
+        self
+    }
+}
+
+enum DrawInstance<'a> {
+    Unhinted(Size, LocationRef<'a>),
+    NativeHinted(&'a NativeHintingInstance),
+}
+
+impl<'a, L> From<(Size, L)> for DrawSettings<'a>
+where
+    L: Into<LocationRef<'a>>,
+{
+    fn from(value: (Size, L)) -> Self {
+        DrawSettings::new(value.0, value.1.into())
+    }
+}
+
+impl From<Size> for DrawSettings<'_> {
+    fn from(value: Size) -> Self {
+        DrawSettings::new(value, LocationRef::default())
+    }
+}
+
+impl<'a> From<&'a NativeHintingInstance> for DrawSettings<'a> {
+    fn from(value: &'a NativeHintingInstance) -> Self {
+        DrawSettings::native_hinted(value)
+    }
+}
+
+/// A scalable glyph outline.
+#[derive(Clone)]
+pub struct OutlineGlyph<'a> {
+    kind: OutlineKind<'a>,
+}
+
+impl<'a> OutlineGlyph<'a> {
+    /// Returns the underlying source format for this outline.
+    pub fn format(&self) -> OutlineGlyphFormat {
+        match &self.kind {
+            OutlineKind::Glyf(..) => OutlineGlyphFormat::Glyf,
+            OutlineKind::Cff(cff, ..) => {
+                if cff.is_cff2() {
+                    OutlineGlyphFormat::Cff2
+                } else {
+                    OutlineGlyphFormat::Cff
+                }
+            }
+        }
+    }
+
+    /// Returns a value indicating if the outline may contain overlapping
+    /// contours or components.
+    ///
+    /// For CFF outlines, returns `None` since this information is unavailable.
+    pub fn has_overlaps(&self) -> Option<bool> {
+        match &self.kind {
+            OutlineKind::Glyf(_, outline) => Some(outline.has_overlaps),
+            _ => None,
+        }
+    }
+
+    /// Returns a value indicating whether the outline has hinting
+    /// instructions.
+    ///
+    /// For CFF outlines, returns `None` since this is unknown prior
+    /// to loading the outline.
+    pub fn has_hinting(&self) -> Option<bool> {
+        match &self.kind {
+            OutlineKind::Glyf(_, outline) => Some(outline.has_hinting),
+            _ => None,
+        }
+    }
+
+    /// Returns the size (in bytes) of the temporary memory required to draw
+    /// this outline.
```

Comment:
I feel like this should point me to what to do with it? Meant to be used to ... or See pattern at blah.

Reply:
Added an explanation, link to the with_memory method that provides more and a table showing how the hinting parameter should be mapped to DrawSettings modes